tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -1101,6 +1101,50 @@ static const ArBasicKind g_AnyCT[] =
   AR_BASIC_UNKNOWN
 };
 
+static const ArBasicKind g_AnyObjectCT[] =
+{
+  AR_OBJECT_SAMPLER,
+  AR_OBJECT_SAMPLERCOMPARISON,
+  AR_OBJECT_TEXTURE1D,
+  AR_OBJECT_TEXTURE1D_ARRAY,
+  AR_OBJECT_TEXTURE2D,
+  AR_OBJECT_TEXTURE2D_ARRAY,
+  AR_OBJECT_TEXTURE3D,
+  AR_OBJECT_TEXTURE2DMS,
+  AR_OBJECT_TEXTURE2DMS_ARRAY,
+  AR_OBJECT_TEXTURECUBE,
+  AR_OBJECT_TEXTURECUBE_ARRAY,
+  AR_OBJECT_STRING_LITERAL,
+  AR_OBJECT_STRING,
```

Comment:
String and String literal are only supported in special cases for constant initializer of subobjects.  I don't think they should be included here.